lib/cli/index.js

```diff
@@ -16,148 +16,105 @@ var chalk = require('chalk');
+ *
+ * @param {Array.<*>|Object} argv - CLI arguments.
  * @param {function(Error?, boolean)} done - Callback
  *   invoked when done.
  */
-function run(cli, isFirstRunOfWatchSession, done) {
+function engine(argv, done) {
+    var cli = new CLI(argv);
     var enabled = chalk.enabled;
+    var watcher;
+
+    chalk.enabled = cli.color;
 
-    var getSpinnerFrame;
     if (cli.watch) {
-        // Spin the spinner!
-        getSpinnerFrame = spinner();
-        var spinnerIntervalId = setInterval(function () {
-            logUpdate(getSpinnerFrame());
-        }, 50);
+        cli.stdout(chalk.bold('Watching...') + ' (press CTRL+C to exit)')
     }
 
-    fileSetPipeline.run(cli, function (err) {
-        /*
-         * The following clearInterval() stops the spinner from further
-         * animation, but it does not clear out the spinner off of the
-         * console.
-         * That part is taken care of in ./file-set-pipeline/log.js .
-         * We don't clear out the spinner here for the following reason.
-         * The way a spinner is cleared out is by erasing the last line in
-         * the console.
-         * If we do that here, the last line of all the plugins' output will
-         * be erased, instead of the spinner.
-         */
-        clearInterval(spinnerIntervalId);
+    run(cli, function (err, success) {
+        chalk.enabled = enabled;
+
+        done(err, success);
 
         /*
-         * Check if any file has failed.
+         * Exit when not-watching, or when an error
+         * has occurred.
          */
 
-        var hasFailed = (cli.files || []).some(function (file) {
-            return (file.messages || []).some(function (message) {
-                return message.fatal === true ||
-                    (message.fatal === false && cli.frail);
-            });
-        });
+        if (err || !cli.watch) {
+            return;
+        }
 
-        chalk.enabled = enabled;
+        /*
+         * Trigger warning when files need to be cached.
+         */
 
-        done(err, !hasFailed);
-    });
+        if (cli.cache.length) {
+            cli.stderr(
+                chalk.yellow('Warning') + ': mdast does not overwrite ' +
+                'watched files until exit.\nMessages and other files are ' +
+                'not affected.'
+            );
+        }
 
-    if (!isFirstRunOfWatchSession) {
         /*
-         * Without the following, the last line of the output from a non-first
-         * pipeline run could get deleted, when ./file-set-pipeline/log.js
-         * tries to clear out the spinner.
-         *
-         * In contrast, if the following is called also on the *first*
-         * pipeline run, then you will see an extra spinner.
+         * Watch files source-locations of files in
+         * the file-set.
          */
-        cli.stdout(getSpinnerFrame());
-    }
+
+        watcher = chokidar.watch(cli.fileSet.sourcePaths, {
+            'ignoreInitial': true
```

Comment:
That’s style, IMO. I always quote props, just so I never have to think about it.